ast/desugar/Desugar.cc

```diff
@@ -474,12 +478,14 @@ TreePtr node2TreeImpl(DesugarContext dctx, unique_ptr<parser::Node> what) {
                     Send::ARGS_store args;
                     unique_ptr<parser::Node> block;
                     args.reserve(send->args.size());
+                    DesugarContext newDctx(dctx.ctx, dctx.uniqueCounter, dctx.enclosingBlockArg,
+                                           dctx.enclosingMethodLoc, dctx.enclosingMethodName, true);
```

Comment:
You're never resetting this to `isInArgs` field to `false` if we're no longer in the context of a sig.

```ruby
# typed: strict
extend T::Sig

sig {params(x: {y: Integer, y: Integer}).void}
def foo(x)
end
```

This says duplicated parameter name but it should say duplicate key.

I think we should give up trying to distinguish between these two cases, because they are actually the same case, and report a duplicate key error in both cases. It will mean that we can do this change without needing to add another boolean.

Reply:
Gould catch! I didn't thought this signature is valid.

Reply:
I've been thinking about reseting `isInArgs` for some time. I guess, there is no easy way to do it. 
One of options is to create new `DesugarContext` for every `node2TreeImpl` call, but it seems like an overkill. 
So I removed `isInArgs` and distinction between hash and kwargs for this error.